File: src/main/java/com/examportal/dto/ExamDTO.java
package com.examportal.dto;

import java.time.LocalDateTime;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class ExamDTO {
    
    @NotBlank(message = "Title must be provided")
    private String title;
    
    @NotBlank(message = "Subject must be provided")
    private String subject;
    
    @NotNull(message = "Total marks must be provided")
    @Positive(message = "Total marks must be positive")
    private Integer totalMarks;
    
    @NotNull(message = "Duration must be provided")
    @Positive(message = "Duration must be positive")
    private Integer durationMin;
    
    @NotNull(message = "Start time must be provided")
    private LocalDateTime startTime;
    
    @NotNull(message = "End time must be provided")
    private LocalDateTime endTime;
}

File: Dotnet3/ExamPortal/DTOs/ExamDTO.cs
using System.ComponentModel.DataAnnotations;

namespace ExamPortal.DTOs
{
    public class ExamDTO
    {
        [Required(ErrorMessage = "Title must be provided")]
        public string Title { get; set; } = string.Empty;
        
        [Required(ErrorMessage = "Subject must be provided")]
        public string Subject { get; set; } = string.Empty;
        
        [Required(ErrorMessage = "Total marks must be provided")]
        [Range(1, int.MaxValue, ErrorMessage = "Total marks must be positive")]
        public int TotalMarks { get; set; }
        
        [Required(ErrorMessage = "Duration must be provided")]
        [Range(1, int.MaxValue, ErrorMessage = "Duration must be positive")]
        public int DurationMin { get; set; }
        
        [Required(ErrorMessage = "Start time must be provided")]
        public DateTime StartTime { get; set; }
        
        [Required(ErrorMessage = "End time must be provided")]
        public DateTime EndTime { get; set; }
    }
}
